cis194/week4/tien/homework4/src/Homework4.hs

```diff
@@ -19,12 +21,12 @@ fun2 n
   | otherwise = fun2 (3 * n + 1)
 
 fun2' :: Integer -> Integer
-fun2' =
-  sum .
-  filter even .
-  takeWhile (> 1) .
-  iterate
-    (\x ->
-       if even x
-         then x `div` 2
-         else 3 * x + 1)
+fun2' = sum . filter even . takeWhile (> 1) . iterate configInputs
+  where
+    configInputs = bool <$> (+ 1) . (* 3) <*> flip div 2 <*> even
```

Comment:
Nice one here. just FYI, the `<$>` here is the same as `.`. So the above line will be the same as:
`configInputs = bool . (+ 1) . (* 3) <*> flip div 2 <*> even`